changhyun/app.js

```diff
@@ -1,36 +1,168 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const express = require('express');
-const cors = require('cors');
-const morgan = require('morgan');
-const { DataSource } = require('typeorm');
+const express = require("express");
+const cors = require("cors");
+const morgan = require("morgan");
+const { DataSource } = require("typeorm");
 const app = express();
 
-const PORT = process.env.PORT
+const PORT = process.env.PORT;
 
 const appDataSource = new DataSource({
   type: process.env.TYPEORM_CONNECTION,
   host: process.env.TYPEORM_HOST,
   port: process.env.TYPEORM_PORT,
   username: process.env.TYPEORM_USERNAME,
   password: process.env.TYPEORM_PASSWORD,
-  database: process.env.TYPEORM_DATABASE
-})
+  database: process.env.TYPEORM_DATABASE,
+});
 
-appDataSource.initialize()
+appDataSource
+  .initialize()
   .then(() => {
-      console.log("Data Source has been initialized!")
+    console.log("Data Source has been initialized!");
   })
   .catch((error) => {
     console.error(error);
-  })
+  });
 
 app.use(cors());
-app.use(morgan('dev'));
+app.use(morgan("dev"));
 app.use(express.json());
 
 app.get("/ping", function (req, res, next) {
-  return res.status(200).json({ message : "pong" });
-})
+  return res.status(200).json({ message: "pong" });
+});
+
+app.post("/user/info", async (req, res, next) => {
+  const { name, email, profileImage, password } = req.body;
+
+  await appDataSource.query(
+    `INSERT INTO users(
+      name,
+      email,
+      profile_image,
+      password
+    ) VALUES (?, ?, ?, ?);
+    `,
+    [name, email, profileImage, password]
+  );
+
+  res.status(201).json({ message: "userCreated" });
+});
+
+app.post("/post", async (req, res, next) => {
+  const { title, content, imageUrl, userId } = req.body;
+
+  await appDataSource.query(
+    `INSERT INTO posts(
+      title,
+      content,
+      image_url,
+      user_id
+    ) VALUES (?, ?, ?, ?);
+      `,
+    [title, content, imageUrl, userId]
+  );
+
+  res.status(201).json({ message: "postCreated" });
+});
+
+app.get("/posts/view", async (req, res, next) => {
```

Comment:
창현님! 해당 API는 모든 게시글을 **_가져오는_** 기능을 하는 API인 것 같네요. 그렇다면 엔드포인트는 `/posts`로 작성되어 http method 중 get과 함께 호출 되면 될 것 같아요! 이와 같이 http method와 url endpoint의 조합으로 표현이 충분히 되기 때문에 굳이 /view라는건 빠져도 좋을 것 같습니다 :)
```suggestion
app.get("/posts", async (req, res, next) => {
```